mintpy/objects/stack.py

```diff
@@ -438,17 +484,23 @@ def get_design_matrix4average_velocity(date_list, refDate=None):
         Parameters: date_list : list of string in YYYYMMDD format
         Returns:    A : 2D array of int in size of (numDate, 2)
         """
-        # convert list of YYYYMMDD into array of diff year in float
-        dt_list = [dt.datetime.strptime(i, '%Y%m%d') for i in date_list]
-        yr_list = [i.year + (i.timetuple().tm_yday - 1) / 365.25 for i in dt_list]
-        yr_diff = np.array(yr_list)
-
+        # convert list of YYYYMMDD into array of years in float
+        date_format = get_date_str_format(date_list[0])
+        dt_list = [dt.strptime(i, date_format) for i in date_list]
+        yr_list = [(d.year + (d.timetuple().tm_yday - 1) / 365.25 + 
+                    d.hour / (365.25 * 24) + 
+                    d.minute / (365.25 * 24 * 60) +
+                    d.second / (365.25 * 24 * 60 * 60))
+                   for d in dt_list]
+        yr_diff = np.array(yr_list, dtype=np.float64)
+
+        # reference date
         if refDate is None:
             refDate = date_list[0]
         yr_diff -= yr_diff[date_list.index(refDate)]
 
-        #for precision, use float32 in 0.1 yr, or float64 in 2015.1 yr format
-        A = np.ones([len(date_list), 2], dtype=np.float32)
+        # use float64 to support very short tbase for UAVSAR data
```

Comment:
A design matrix in float64 will slow down the least-squares inversion by half. We should use float32 instead, following the unit of days instead of the unit of years.